core/src/main/java/io/questdb/cutlass/http/processors/ExportQueryProcessor.java

```diff
@@ -193,9 +189,18 @@ public void execute(
             if (state.recordCursorFactory != null) {
                 try {
                     boolean runQuery = true;
+                    boolean canStreamExportParquet = state.getExportModel().isParquetFormat() && CopyExportContext.canStreamExportParquet(state.recordCursorFactory);
+                    final int order = state.recordCursorFactory.getScanDirection() == SCAN_DIRECTION_BACKWARD ? ORDER_DESC : ORDER_ASC;
+                    state.descending = order == ORDER_DESC;
                     for (int retries = 0; runQuery; retries++) {
                         try {
-                            state.cursor = state.recordCursorFactory.getCursor(sqlExecutionContext);
+                            if (canStreamExportParquet) {
+                                state.recordCursorFactory.changePageFrameSizes(state.getExportModel().getRowGroupSize(), state.getExportModel().getRowGroupSize());
```

Comment:
this is sub-optimal and leads to side effects. factories are cached. Now we change row group side on the factory and return it to cache. Next time factory is executed with wrong defaults - we need a test to that end, actually. Consider taking row group sizes from execution context rather than lateral factory mutation.